core/src/test/java/dk/alexandra/fresco/framework/network/async/TestAsyncNetwork.java

```diff
@@ -0,0 +1,316 @@
+package dk.alexandra.fresco.framework.network.async;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import dk.alexandra.fresco.framework.Party;
+import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
+import dk.alexandra.fresco.framework.configuration.NetworkConfigurationImpl;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.nio.channels.ServerSocketChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import javax.net.ServerSocketFactory;
+import org.junit.Test;
+
+public class TestAsyncNetwork {
+
+  private List<Integer> getFreePorts(int no) {
+    List<Integer> res = new ArrayList<>();
+    for (int i = 0; i < no; i++) {
+      res.add(9000 + i);
+    }
+    return res;
+  }
+
+  @Test(timeout = 5000)
+  public void testSendBytes() {
+    Map<Integer, Party> parties = new HashMap<>();
+    List<Integer> ports = getFreePorts(2);
+    parties.put(1, new Party(1, "localhost", ports.get(0), "78dbinb27xi1i"));
+    parties.put(2, new Party(2, "localhost", ports.get(1), "h287hs287g22n"));
```

Comment:
There is a lot of boiler-plate setup in these tests. Please refactor to make the tests more readable.

Reply:
I.e., setting up parties and networks, could probably be factored out nicely for most of these tests.